drivers/ethernet/eth_mcux.c

```diff
@@ -283,11 +283,16 @@ static inline struct net_if *get_iface(struct eth_context *ctx, u16_t vlan_tag)
 static void eth_mcux_phy_enter_reset(struct eth_context *context)
 {
 	/* Reset the PHY. */
+#ifndef CONFIG_ETH_MCUX_NO_PHY_SMI
```

Comment:
For all the ifdefs in this patch, looks like you could use if (IS_ENABLED(...)) instead. (or if (!IS_ENABLED(...)) here)